opencog/rule-engine/forwardchainer/ForwardChainer.cc

```diff
@@ -124,10 +125,19 @@ void ForwardChainer::do_step(void)
     _log->debug( "Derived rule size = %d", derived_rhandles.size());
 
     HandleSeq products;
-    //Applying all partial groundings found.
+    //Applying all partial/full groundings.
     for (Handle rhandle : derived_rhandles) {
-        HandleSeq hs = apply_rule(rhandle,_search_focus_Set);
-        products.insert(products.end(),hs.begin(),hs.end());
+        HandleSeq hs;
+        //Check for fully grounded outputs returned by derive_rules.
+        if (not contains_atomtype(rhandle, VARIABLE_NODE)) {
```

Comment:
Its the implicand part which is checked https://github.com/opencog/atomspace/blob/master/opencog/rule-engine/forwardchainer/ForwardChainer.cc#L594 . So if implicand doesn't have variables after partial grounding, just implicand part is pushed to the derived rules vector and returned since I wasn't able to create able with no variables inside it.